Python/uniqueid.c

```diff
@@ -90,10 +91,15 @@ _PyObject_AssignUniqueId(PyObject *obj)
     }
 
     _Py_unique_id_entry *entry = pool->freelist;
+    Py_ssize_t unique_id = (entry - pool->table);
+    if ((uint64_t)unique_id >= DICT_UNIQUE_ID_MAX) {
```

Comment:
Does it really make sense to limit all unique ids just because dicts can only store (currently) 51 bits worth of unique ids? (It's also a bit meh how far away this check is from the dict implementation, but that goes for the existing fiddling with code/type object internals in clear_unique_id as well.) Can't we just check the bounds of unique_id in _PyDict_EnablePerThreadRefcounting instead, and release the uniqueid if it's too big to fit?

Reply:
I moved the check to `_PyDict_EnablePerThreadRefcounting`